image-inspector.go

```diff
@@ -129,55 +185,45 @@ func getAuthConfigs(dockercfg, username, password_file *string) *docker.AuthConf
 			log.Fatalf("No auths were found in the given dockercfg file\n")
 		}
 	}
-	if *username != "" {
-		token, err := ioutil.ReadFile(*password_file)
+	if username != "" {
+		token, err := ioutil.ReadFile(password_file)
 		if err != nil {
 			log.Fatalf("Unable to read password file: %v\n", err)
 		}
 		imagePullAuths = &docker.AuthConfigurations{
-			map[string]docker.AuthConfiguration{"": docker.AuthConfiguration{Username: *username, Password: string(token)}}}
+			map[string]docker.AuthConfiguration{"": docker.AuthConfiguration{Username: username, Password: string(token)}}}
 	}
 
 	return imagePullAuths
 }
 
 func main() {
-	uri := flag.String("docker", "unix:///var/run/docker.sock", "Daemon socket to connect to")
-	image := flag.String("image", "", "Docker image to inspect")
-	dstpath := flag.String("path", "", "Destination path for the image files")
-	serve := flag.String("serve", "", "Host and port where to serve the image with webdav")
-	chroot := flag.Bool("chroot", false, "Change root when serving the image with webdav")
-	dockercfg := flag.String("dockercfg", "", "Location of the docker configuration file")
-	username := flag.String("username", "", "username for authenticating with the docker registry")
-	password_file := flag.String("password-file", "", "Location of a file that contains the password for authentication with the docker registry")
+	inspector := NewDefaultImageInspectorOptions()
```

Comment:
@pweil- maybe we want to evaluate the possibility to move this all out of main in a dedicated function (different file as well if we want to keep things tight).

No need for you to update this PR now. Just a suggestion going forward.